CONTRIBUTING.md

```diff
@@ -0,0 +1,196 @@
+
+- The **role of Model** is to store the activity logs from background and store the filters. It also checkes if logs are matching or not matching with the applied filters.
+
+- The **role of View** is to deal with DOM. It interacts with DOM whenver our Model is updated. The `View` deals with `log-view` web component to render the logs and [filter web components](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component) to display filters.
+
+- The **role of Controller** is to make sure that both Model and View are synchronized. Whenever the Model updates with new logs or filter options, the Controller make sure that the View also updates accordingly.
+
+When the Activity Log page is opened via popup, it fetches the existing logs (if logs were collected before) from the background. The fetched logs are saved in the [**Model**](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-activitylog.js#L5), then **View** renders the logs with the help of [`log-view`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/web-component/log-view/) web component.
+
+While the Activity Log page is opened, it receives real-time logs from background and render them. It can send instructions in the background to save logs to a JSON file, load logs from JSON file and clear logs.
+
+It also provides the functionality to filter out the unncessary logs by log identities, substring searching and with tab id. These filtering options are made with a couple of [web components](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/web-component). The filters are stored in the [**Model**](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-activitylog.js#L6-L14) and the **View** updates the `log-view` everytime the filters are updated.
+
+#### Activity Log Page - Devtools (Extension Page)
+
+This page can be accessed via "Extension Activity" panel in devtools. The "Extension Activity" panel contains the Activity Log page. It has almost all the similar characteristics as "Activity Log Page - Tab" except it filters the activity logs with tab's id where devtools panel is opened. The tab id is found with the help of devtools API i.e. `browser.devtools.inspectedWindow.tabId`.
+
+### Core Features
+
+#### Collecting Logs
+
+The extension receives each activity log in the form of an `object` from activityLog API. The activityLog API schema can been found [here](https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/activity_log.json). The background page (backend part) subscribes the `browser.activityLog.onExtensionActivity` API event listener, which gets triggered and returns a log object everytime an activity is found from any monitored extension. While monitoring other extensions, any newly installed extension is being monitored automatically.
```

Comment:
it is technically incorrect to say that "the onExtensionActivity event listener returns a log object ...", the event listener is called with a activity log object passed as a parameter.